### PR TITLE
Fixes creation of a vis from a table

### PR DESCRIPTION
**Pull request by javierarce**

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1506/))

**Comment by javisantana** [comment]
:es: I would explain why we use master_vis instead of vis in a comment

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1514/))

**Comment by javisantana** [comment]
please update news if needed and merge

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1741/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1757/))

**Comment by xavijam** [comment]
And this pull request?

**Comment by javierarce** [comment]
I need to merge it with master and release it.

**Comment by javierarce** [comment]
I've just tested it in staging, it's throwing an error:

`Uncaught Error: A "url" property or function must be specified
cdb.js:3 GET https://xxx.cartodb-staging.com/u/[username]/api/v1/maps/c6dc8059-7b48-4951-ad8a-b6df6972dad8/layers 404 (Not Found)``

This can be reproduced as follows:

1. Create an empty table.
2. Open it and go to the map view.
3. Click on the + sing to add a new layer.
4. Pick a table from the list and add it.
5. Add a name for the visualization.

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2085/))

**Comment by javisantana** [comment]
ping?

**Comment by xavijam** [comment]
And in local was working?

**Comment by javierarce** [comment]
@xavijam: no, after another testing in local it's throwing a js error.

After creating the vis, it fails in the parsing here: https://github.com/CartoDB/cartodb/blob/master/lib/assets/javascripts/cartodb/models/cartodb_layer.js#L267

because `data` is null.

If I exit the method in the case that var is null, I get this one: 

`Uncaught Error: A "url" property or function must be specified cdb.js:3 GET https://xxx.cartodb-staging.com/u/[username]/api/v1/maps/c6dc8059-7b48-4951-ad8a-b6df6972dad8/layers 404 (Not Found)`


I'm stuck :|

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2222/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2231/))

File: lib/assets/javascripts/cartodb/models/cartodb_layer.js [hunk 1]
```diff
@@ -259,7 +259,7 @@ cdb.admin.CartoDBLayer = cdb.geo.CartoDBLayer.extend({
     var c = {};
     // in case this is a response of saving the layer discard
     // values from the server
-    if (this._saving && !this.isNew()) {
+    if (!data || this._saving && !this.isNew()) {
```
Comment (javisantana) [inline lib/assets/javascripts/cartodb/models/cartodb_layer.js]: this probably fixes it but this but i don't think this is the source of the problem

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2232/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2233/))

**Comment by javierarce** [comment]
@javisantana: can you have a look to the [last changes](https://github.com/CartoDB/cartodb/commit/06ffa64f33f5a844502857c3efe8753f491499d8)?

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2267/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2268/))

**Comment by javisantana** [comment]
:es: :fear:

**Pull request merged**